src/bin/pgcopydb/pgsql.c

```diff
@@ -2849,9 +2854,9 @@ pg_copy_data(PGSQL *src, PGSQL *dst, CopyArgs *args)
 		/*
 		 * If successful PQgetCopyData returns the row length as a result.
 		 */
-		else if (bufsize > 0)
+		else if (bufsize > 0 && on_copy_progress_hook != NULL)
 		{
-			args->bytesTransmitted += bufsize;
+			on_copy_progress_hook(bufsize, context);
```

Comment:
The callback function prototype returns a boolean value to indicate success or error. We need to process that return value here. Then there is a choice of logging a WARN/DEBUG message or an ERROR message (plus stopping the COPY operation) in case the callback returns false, but that needs to be explicit.

Reply:
done